core/ssz.go

```diff
@@ -0,0 +1,380 @@
+// Copyright © 2022-2023 Obol Labs Inc. Licensed under the terms of a Business Source License 1.1
+
+package core
+
+import (
+	"testing"
+
+	eth2bellatrix "github.com/attestantio/go-eth2-client/api/v1/bellatrix"
+	eth2capella "github.com/attestantio/go-eth2-client/api/v1/capella"
+	eth2spec "github.com/attestantio/go-eth2-client/spec"
+	"github.com/attestantio/go-eth2-client/spec/altair"
+	"github.com/attestantio/go-eth2-client/spec/bellatrix"
+	"github.com/attestantio/go-eth2-client/spec/capella"
+	"github.com/attestantio/go-eth2-client/spec/deneb"
+	eth2p0 "github.com/attestantio/go-eth2-client/spec/phase0"
+	ssz "github.com/ferranbt/fastssz"
+	"github.com/stretchr/testify/require"
+
+	"github.com/obolnetwork/charon/app/errors"
+	"github.com/obolnetwork/charon/app/z"
+)
+
+// sszType indicates a type that can be marshalled and unmarshalled by ssz.
+type sszType interface {
+	ssz.Marshaler
+	ssz.Unmarshaler
+}
+
+// ======================= VersionedSignedBeaconBlock =======================
+
+// MarshalSSZ ssz marshals the VersionedSignedBeaconBlock object.
+func (b VersionedSignedBeaconBlock) MarshalSSZ() ([]byte, error) {
```

Comment:
why don't we also move other ssz marshal unmarshal methods from signed data to this file ssz.go?

Reply:
ssz.go is only our own custom implementations, these using go-eth2-client implementations.